src/main/java/org/folio/rest/impl/SamlAPI.java

```diff
@@ -98,11 +113,19 @@ public void getSamlCheck(RoutingContext routingContext, Map<String, String> okap
   public void postSamlLogin(SamlLoginRequest requestEntity, RoutingContext routingContext, Map<String, String> okapiHeaders,
                             Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
 
+    String csrfToken = UUID.randomUUID().toString();
+    Cookie csrfCookie = Cookie.cookie(CSRF_TOKEN, csrfToken)
+      .setPath("/").setHttpOnly(true).setSecure(true);
+    routingContext.addCookie(csrfCookie);
+
     String stripesUrl = requestEntity.getStripesUrl();
+    StringBuilder relayInfo = new StringBuilder(stripesUrl);
+    relayInfo.append(stripesUrl.indexOf('?') >= 0 ? '&' : '?')
+      .append(CSRF_TOKEN).append("=").append(csrfToken);
```

Comment:
Using + is much easier to read.
The compiler automatically uses a StringBuilder internally when we use + in the code.
`String relayInfo = stripesUrl + (stripesUrl.indexOf('?') >= 0 ? '&' : '?') + CSRF_TOKEN + '=' + csrfToken;`

Reply:
> Using + is much easier to read.
> The compiler automatically uses a StringBuilder internally when we use + in the code.
> `String relayInfo = stripesUrl + (stripesUrl.indexOf('?') >= 0 ? '&' : '?') + CSRF_TOKEN + '=' + csrfToken;`

Done.